packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -178,6 +181,22 @@ export default class Extension {
     return true;
   }
 
+  private flushAccountCache (): void {
+    this.cachedAccount = undefined;
```

Comment:
On flush you need to cancel the timeout. Otherwise use may create 1, start the time, start a new one and othe original timer clears.

Reply:
Thanks I added it. I've set the timeout in the state then, that looks a little wild though, wondering if I should rather declare those variables in the `./State`?

Reply:
Will take a look. But my knee-jerk was that it is messy to manage, since you need to track the timerId as well. Don't see another solution though.